modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/KnowledgeBaseArticleResourceImpl.java

```diff
@@ -493,6 +494,12 @@ private KnowledgeBaseArticle _addKnowledgeBaseArticle(
 				KBFolderConstants.DEFAULT_PARENT_FOLDER_ID;
 		}
 
+		Date datePublished = knowledgeBaseArticle.getDatePublished();
+
+		if (datePublished == null) {
+			datePublished = new Date();
```

Comment:
Does the method `addKBArticle` accept a `null` publish date? In this case I would pass `null` instead of a fake date

Reply:
no, the validator of the displayDate throw an exception if the displayDate is null.
Display date must be a date.